src/evaluate/inspect.py

```diff
@@ -31,41 +32,67 @@ class SplitsNotFoundError(ValueError):
     pass
 
 
-def list_metrics(with_community_metrics=True, with_details=False):
+def list_evaluation_modules(type=None, include_community=True, with_details=False):
     """List all the metrics script available on the Hugging Face Hub.
 
     Args:
-        with_community_metrics (:obj:`bool`, optional, default ``True``): Include the community provided metrics.
-        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the short name.
+        type (:obj:`string`, optional, default ``None``): Type of evaluation modules to list. If ``None`` all types are listed.
+        include_community (:obj:`bool`, optional, default ``True``): Include community modules in the list.
+        with_details (:obj:`bool`, optional, default ``False``): Return the full details on the metrics instead of only the ID.
+
+    Returns:
+        `list`
```

Comment:
```suggestion
        `List[str]`
```

Reply:
Although actually this is a `List[Union[str, ...]]` since if `with_details=True` return more info.